crates/ruff_linter/src/rules/pyupgrade/rules/use_pep604_annotation.rs

```diff
@@ -6,16 +6,17 @@ use ruff_python_semantic::analyze::typing::Pep604Operator;
 use ruff_text_size::Ranged;
 
 use crate::checkers::ast::Checker;
+use crate::codes::Rule;
 use crate::fix::edits::pad;
 use crate::settings::types::PythonVersion;
 
 /// ## What it does
-/// Check for type annotations that can be rewritten based on [PEP 604] syntax.
+/// Check for `typing.Union` annotations that can be rewritten based on [PEP 604] syntax.
```

Comment:
We should keep the documentation as is and instead add a Preview section that mentions that the rule now only covers `typing.Union` and that the `typing.Optional` is covered by a new rule